core/services/workflows/engine.go

```diff
@@ -649,13 +649,13 @@ func (e *Engine) finishExecution(ctx context.Context, cma custmsg.MessageEmitter
 	if exists {
 		// send metering report to beholder
 		if err = events.EmitMeteringReport(ctx, cma, report.Message()); err != nil {
-			e.metrics.with(platform.KeyWorkflowID, e.workflow.id, platform.KeyWorkflowExecutionID, executionID).incrementWorkflowMissingMeteringReport(ctx)
+			e.metrics.with(platform.KeyWorkflowID, e.workflow.id, executionID).incrementWorkflowMissingMeteringReport(ctx)
```

Comment:
```suggestion
			e.metrics.incrementWorkflowMissingMeteringReport(ctx)
```

since WorkflowID is static, we should set that label on the metrics object when the engine is first initialized.

Reply:
Added on line 158